src/version.rs

```diff
@@ -0,0 +1,934 @@
+use std::borrow::Cow;
+use std::cmp::Ordering;
+use std::fmt;
+
+/// A full RPM "NEVRA" consists of 5 different components - Name, Epoch, Version, Release, and Architecture.
+///
+/// Name is the name of the package.
+///
+/// Epoch overrides all other fields and is generally only used as a last resort - in cases where
+/// a change to the versioning scheme or packaging error creates a situation where newer packages
+/// might otherwise sort as being older.
+///
+/// Version is the normal version string used by the upstream project. This shouldn't be tweaked
+/// by the packager.
+///
+/// Release indicates firstly the number of times this package has been released - for instance,
+/// with custom patches and backports not present in the upstream, but may also indicate other
+/// details such as the OS it was built for (fc38, el9) or portions of a git commit hash.
+///
+/// Architecture indicates the CPU architecture that this package is intended to support.
+///
+/// In many contexts (on a system, in a repository), package NEVRAs are meant to be unique. You can have
+/// different packages with the same NEVRA - but you can't install both, or put them both in a repo.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct NEVRA<'a> {
```

Comment:
Can we avoid all caps types?

Reply:
NEVRA is an acronym, is the main reason.  I can change it, but then for instance there will be a difference between the name as used in the docs (where all caps is standard for such acronyms) and the type name.

Reply:
That's fine, acronyms are either to be camel (so `Nevra`) cased or named to something that can be interpreted, i.e. `PackageVersionFull` or similar.